### DISCUSSION TITLE
Will Zed aim for configuration to be scriptable a'la Lua/Vimscript/Lisp?

### DISCUSSION THREAD

**Discussion by tkgalk** [zed-industries/zed#11745] [Feature Requests]
One of the absolutely best parts of the truly hackable editors is a good API with the possibility to write your configuration as a script. This isn't possible currently in Zed (AFAIK) and it seems like the JSON configurations approach was chosen.

I'm wondering why was this approach chosen and if this could change in the future (as an opt-in).

**Comment by habruening** [comment]
I made a detailed proposal for Lisp. See here: https://github.com/zed-industries/zed/discussions/14539

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
